### PR TITLE
Implicit returns

### PR DESCRIPTION
**Pull request by sornas**

### CODE REVIEW

File: sylt-parser/src/expression.rs [hunk 1]
```diff
@@ -146,9 +146,36 @@ fn function<'t>(ctx: Context<'t>) -> ParseResult<'t, Expression> {
...
     // Parse the function statement.
-    let (ctx, statement) = block_statement(ctx)?;
+    let (ctx, mut statement) = block_statement(ctx)?;
+
+    // If the return type isn't void, check for and apply implicit returns.
+
+    let statements = if let StatementKind::Block { statements } = &mut statement.kind {
+        statements
+    } else {
+        unreachable!("Function blocks should only be blocks");
```
Comment (FredTheDino) [inline sylt-parser/src/expression.rs:157]: Do we want to allow expressions? :o

File: progs/tests/implicit_return/simple.sy [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+h :: fn a:int, b:int -> int {
+    a + b
+}
+
+start :: fn {
+    f() <=> 1
+    g() <=> 2
+    h(1, 2) <=> 3
+}
```
Comment (FredTheDino) [inline progs/tests/implicit_return/simple.sy:17]: Maybe add more tests?

**Review by FredTheDino — APPROVED** [review]
LGTM

File: sylt-parser/src/expression.rs (hunk 1 above)
Comment (sornas) [reply sylt-parser/src/expression.rs:157]: Maybe? Not here at least

**Pull request merged**